include/pybind11/numpy.h

```diff
@@ -287,6 +282,7 @@ struct npy_api {
         void **api_ptr = (void **) PyCapsule_GetPointer(c.ptr(), nullptr);
         if (api_ptr == nullptr) {
             raise_from(PyExc_SystemError, "FAILURE obtaining numpy _ARRAY_API pointer.");
+            throw error_already_set();
```

Comment:
I think you're right! This needs to be here. How did you notice?

Looks like a made a mistake in #4570.

Reply:
Here in documentation: https://pybind11.readthedocs.io/en/stable/advanced/exceptions.html#chaining-exceptions-raise-from
> To do a similar thing in pybind11, you can use the py::raise_from function. It sets the current python error indicator, so to continue propagating the exception you should throw py::error_already_set().

Reply:
Good catch (it completely slipped my mind), thanks a lot!